lib/WPG/Service/Payment/Notification.php

```diff
@@ -0,0 +1,500 @@
+<?php
+
+namespace ThanhVo\Worldpay\WPG\Service\Payment;
+
+class Notification
+{
+
+    /**
+     * @var mixed
+     */
+    private $orderStatusEvent;
+
+    /**
+     * @var mixed
+     */
+    private $payment;
+
+    /**
+     * @var mixed
+     */
+    private $journal;
+
+    /**
+     * @var string
+     */
+    private $lastEvent;
+
+    /**
+     * Notification constructor.
+     * @param string $response
+     */
+    public function __construct(string $response)
+    {
+        $data = new \SimpleXMLElement($response);
+        $this->orderStatusEvent = $data->notify->orderStatusEvent;
+        $this->payment = $this->orderStatusEvent->payment;
+        $this->journal = $this->orderStatusEvent->journal;
+        $this->lastEvent = $this->payment->lastEvent->__toString();
+    }
```

Comment:
There should be PaymentStatus class, then migrate all status related methods into.
```php
class PaymentStatus
{
    public function isShopperCancelled(): bool
    {
        ...
    }
    ...
}

class Notification
{
    ...
    public function __construct(string $response)
    {
        $this->lastEvent = new PaymentStatus($this->payment->lastEvent->__toString());
        $this->journalType = new PaymentStatus($this->orderStatusEvent->journal['journalType']->__toString());
    }
    ...
    public getLastEvent(): Payment
    {
        return $this->lastEvent;
    }
    ...
    public getJournalType(): Payment
    {
        return $this-> journalType;
    }
}